dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportService.java

```diff
@@ -209,6 +211,15 @@ public Map<Class<? extends IdentifiableObject>, List<? extends IdentifiableObjec
     return metadata;
   }
 
+  /**
+   * This predicate is used to filter out deprecated Analytics schemas, {@link EventChart} & {@link
+   * EventReport}.As they are no longer used ({@link EventVisualization} has replaced them), they
+   * should be removed from the metadata export flow. This was causing issues otherwise. See <a
+   * href="https://dhis2.atlassian.net/browse/BETA-177">Jira issue</a>
+   */
+  public static final Predicate<Schema> DEPRECATED_ANALYTICS_SCHEMAS =
```

Comment:
I assume it can be private?
I also would make this a constant for `Set<Class<?>>` instead and use that set in the filter. Seems less overly specific.

Reply:
It could be private. I wanted to add tests for it. I don't see any issue with it being public. It doesn't reveal anything unwanted about the class.
I created the predicate so tests could be added to cover the logic applied. This seems more valuable than using an inline set in the filter.